tests/ContextTest.py

```diff
@@ -95,5 +97,103 @@ def test_add_import(self):
+        stmt2 = create_mock_statement(ident_uri, 2)
+        stmt3 = create_mock_statement(ident_uri, 3)
+        stmt4 = create_mock_statement(ident_uri, 4)
+        [ctx.add_statement(stmt) for stmt in [stmt1, stmt2, stmt3, stmt4]]
+        ctx.remove_statement(stmt2)
+        self.assertTrue(len(list(ctx.contents_triples())), 3)
+
+    def test_add_bad_statement(self):
+        ctx = Context(ident='http://example.com/context_1')
+        stmt1 = create_mock_statement('http://example.com/context_2', 1)
+        with self.assertRaises(ValueError):
+            ctx.add_statement(stmt1)
+
+    def test_contents_triples(self):
+        ident_uri = 'http://example.com/context_1'
+        ctx = Context(ident=ident_uri)
+        stmt1 = create_mock_statement(ident_uri, 1)
+        stmt2 = create_mock_statement(ident_uri, 2)
+        stmt3 = create_mock_statement(ident_uri, 3)
+        stmt4 = create_mock_statement(ident_uri, 4)
+        [ctx.add_statement(stmt) for stmt in [stmt1, stmt2, stmt3, stmt4]]
+        self.assertTrue(all(ctx.contents_triples()))
+
+    def test_clear(self):
+        ident_uri = 'http://example.com/context_1'
+        ctx = Context(ident=ident_uri)
+        for i in range(5):
+            ctx.add_statement(create_mock_statement(ident_uri, i))
+        ctx.clear()
+        self.assertEqual(len(list(ctx.contents_triples())), 0)
+
+    def test_save_context(self):
+        ident_uri = 'http://example.com/context_1'
+        ctx = Context(ident=ident_uri)
+        for i in range(5):
+            ctx.add_statement(create_mock_statement(ident_uri, i))
+        graph = ctx.save_context(set())
+        self.assertEqual(ctx.tripcnt, 5)
+        self.assertEqual(len(graph), 5)
+
+    def test_save_context_with_inline_imports(self):
+        ident_uri = 'http://example.com/context_1'
+        ident_uri2 = 'http://example.com/context_2'
+        ident_uri2_1 = 'http://example.com/context_2_1'
+        ident_uri3 = 'http://example.com/context_3'
+        ident_uri4 = 'http://example.com/context_4'
+        ctx = Context(ident=ident_uri)
+        ctx2 = Context(ident=ident_uri2)
+        ctx2_1 = Context(ident=ident_uri2_1)
+        ctx.add_import(ctx2)
+        ctx.add_import(ctx2_1)
+        ctx3 = Context(ident=ident_uri3)
+        ctx3.add_import(ctx)
+        last_ctx = Context(ident=ident_uri4)
+        last_ctx.add_import(ctx3)
+        ctx.add_statement(create_mock_statement(ident_uri, 1))
+        ctx2.add_statement(create_mock_statement(ident_uri2, 2))
+        ctx2_1.add_statement(create_mock_statement(ident_uri2_1, 2.1))
+        ctx3.add_statement(create_mock_statement(ident_uri3, 3))
+        last_ctx.add_statement(create_mock_statement(ident_uri4, 4))
+        graph = last_ctx.save_context(set(), True)
+        self.assertEqual(len(graph), 5)
+
+    def test_init_bool(self):
+        ident_uri = 'http://example.com/context_1'
+        ctx = Context(ident=ident_uri)
+        self.assertFalse(ctx)
```

Comment:
should be assertTrue -- contexts are always True.

Reply:
oh ok